### PR TITLE
Add mouse region for InputDecorationIcons in the textfield

### LINKED ISSUE #112048: Text insertion mouse cursor should change when hovering InputDecorationIcons
It's easy enough to try this out here:
https://flutter-experimental-m3-demo.web.app/#/

If you hover the leading/trailing icons in the textfield demos, the cursor remains a text insertion bar.

### PR DESCRIPTION
**Pull request by hannah-hyj**
Fixed: #112048

Add mouse region for InputDecorationIcons in the textfield. 

The mouse cursor will be "basic" on the basic icons and "click" on icons buttons. Test case updated.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: bin/internal/engine.version [hunk 1]
```diff
@@ -1 +1 @@
-38fef0572fa06ecf01ee2a1bf56623f7940e4853
+a64c26d44af32401d041729dcca9510011adeeef
```
Comment (HansMuller) [inline bin/internal/engine.version:1]: I don't think this should be part of the commit

**Review by HansMuller — APPROVED** [review]
LGTM except for the engine.version change

**Review by darrenaustin — APPROVED** [review]
The code changes LGTM. It does look like there is several test failures that appear related to this change that will need to be resolved.

**Comment by hangyujin** [comment]
Sorry for the mess, I am closing this PR and creating a new one

**Pull request closed without merging**